packages/dds/matrix/src/matrix.ts

```diff
@@ -435,8 +425,13 @@ export class SharedMatrix<T = any>
 		currentVector: PermutationVector,
 		oppositeVector: PermutationVector,
 		target: SnapshotPath.rows | SnapshotPath.cols,
-		message: IMergeTreeOp,
+		message: IMergeTreeOp | undefined,
```

Comment:
why allow undefined, and then assert it is not undefined. seem better to disallow undefined and force the caller not to pass it

Reply:
i think this change makes the code harder to understand, than keeping the previous asserts.

Reply:
I made this change before there were any checks whatsoever (and code dealt with `any`) - see this change https://github.com/microsoft/FluidFramework/commit/9579629b8d26ea04b5c274fea8425fb14ccaf5f4#diff-87d20d90ea92ec1d7e133b1cc0e9c22433273248e44635baa75e528525f2332fR456
undefined was passed around.

No strong preference on my part, other maybe - less code in final bundle :)

Reply:
i'd prefer it was changed back, to keep the code clear that undefined isn't allowed be submitted. i think its worth the bytes in the bundles

Reply:
i just fixed the problem all up, no more undefined on this path now:
https://github.com/microsoft/FluidFramework/pull/20009

Reply:
which ever goes in first can just merge with the other, so approving